### PR TITLE
Mes 8240 eco reason capture

### PR DESCRIPTION
**Pull request by gittins7**
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Screenshot 2022-10-14 at 15 11 45
Screenshot 2022-10-14 at 15 11 51
Screenshot 2022-10-14 at 15 11 59
Screenshot 2022-10-14 at 15 12 05
Screenshot 2022-10-14 at 15 12 16
Screenshot 2022-10-14 at 15 12 30

### CODE REVIEW

File: src/app/pages/office/cat-adi-part2/components/eco-capture-reason/eco-capture-reason.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<ion-row [formGroup]="formGroup" class="mes-validated-row mes-row-separator-top">
+    <div class="validation-bar" [class.ng-invalid]="invalid"></div>
+    <ion-col class="component-label ion-align-self-start" size="32">
+        <ion-row class="spacing-row"></ion-row>
+        <label>Reason for ECO capture</label>
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/components/eco-capture-reason/eco-capture-reason.html]: please add appropriate ID

File: src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<ion-row [formGroup]="formGroup" class="mes-validated-row">
+    <div class="validation-bar" [class.ng-invalid]="invalid"></div>
+    <ion-col class="component-label ion-align-self-start" size="32">
+        <ion-row class="spacing-row"></ion-row>
+        <label>Fault related to ECO capture</label>
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html]: Same as above

File: src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            <ion-select id="eco-fault-selector"
+                        okText="Submit"
+                        cancelText="Cancel"
+                        placeholder="Select fault"
+                        class="black-text"
+                        formControlName="ecoRelatedFault"
+                        [class.ng-invalid]="invalid"
+                        (ionChange)="ecoFaultChanged($event.target.value)"
+                        [interfaceOptions]="{cssClass:'mes-select', enableBackdropDismiss:false}">
+                <ion-select-option *ngFor="let fault of drivingFaults"
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html]: We could use a trackByIndex which is a recommended on ngFor loops for performance purposes

File: src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                                   [value]="fault.competencyIdentifier">
+                    {{fault.competencyDisplayName}}
+                </ion-select-option>
+            </ion-select>
+        </ion-row>
+        <ion-row class="validation-message-row ion-align-items-center">
+            <div class="validation-text" [class.ng-invalid]="invalid" id="show-me-question-validation-text">
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/components/eco-related-fault/eco-related-fault.html]: please update ID

File: src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts [hunk 5]
```diff
@@ -271,4 +312,17 @@ export class OfficeCatADI2Page extends OfficeBasePageComponent implements OnInit
     this.store$.dispatch(ShowMeQuestionSelected(result, index));
   }
 
+  ecoFaultChanged(fault: string): void {
+    this.store$.dispatch(AddEcoRelatedFault(fault));
+  }
+
+  ecoCaptureReasonChanged(ecoCaptureReason: string): void {
+    console.log('ecoCaptureReason:', ecoCaptureReason);
```
Comment (matthew2564) [inline src/app/pages/office/cat-adi-part2/office.cat-adi-part2.page.ts]: Please remove log

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**